lib/client/redirect.go

```diff
@@ -203,6 +212,62 @@ func (rd *Redirector) callback(w http.ResponseWriter, r *http.Request) (*auth.SS
 	return re, nil
 }
 
+// DELETE IN: 4.1.0
+//
+// open will first attempt to decrypt with Teleport secret package, and if it
+// fails, fallback to legacy lemma package.
+func (rd *Redirector) open(ciphertext []byte) ([]byte, error) {
+	plaintext, err := rd.key.Open(ciphertext)
+	if err != nil {
```

Comment:
does this do some integrity check to make sure it decodes random text, I'm pretty sure it does, but want to confirm

Reply:
It is, AES-GCM is a AEAD cipher.